paper_sections/02-gams.Rmd

```diff
@@ -37,39 +43,52 @@ dat$x <- dat$x2
+ylims <- range(p, p.0, p.inf)
 
-There are many possible basis functions that can be used to model the $b_k$s. Here we'll use thin plate regression splines, which have the appealing property that knot choice is somewhat automatic (the best approximation to including knots at each data poiint is used; @wood_thin_2003).
+lab.l <- labs(x = "x", y = "y")
+dat.l <- geom_point(data = dat, aes(x = x, y = y), colour = "darkgrey")
+true.l <- geom_line(data = true, aes(x = x, y = y), colour = "blue")
+coord.l <- coord_cartesian(ylim = ylims)
 
+p1 <- ggplot(p, aes(x = x, y = fit)) +
+  dat.l + true.l +
+  geom_line(colour = "darkred") + lab.l + coord.l
+
+p2 <- ggplot(p.0, aes(x = x, y = fit)) +
+  dat.l + true.l +
+  geom_line(colour = "darkred") + lab.l + coord.l
+
+p3 <- ggplot(p.inf, aes(x = x, y = fit)) +
+  dat.l + true.l +
+  geom_line(colour = "darkred") + lab.l + coord.l
+
+plot_grid(p1, p2, p3, align = "hv", axis = "lrtb", ncol = 3, labels = "auto")
+```
+
+The number of basis functions, $K$, limits the maximum complexity for a given smooth term. To measure the complexity of a smooth term, we use the *effective degrees of freedom* (EDF), which at a maximum is the number of coefficients to be estimated in the model, minus any constraints. The EDF can take non-integer values and larger values indicate more wiggly terms. See Wood [-@Wood:2006vg, Section 4.4] for further details.
+
+There are many possible basis functions that can be used to construct the $b_k$. Here we'll use thin plate regression splines (TPRS), which have the appealing property that knot choice is somewhat automatic (the best approximation to including knots at each data point is used; [@wood_thin_2003]).
 
 **DLM:: put the cubic splines back in here**
 
-TPRS are also defined for any number of predictors, so multivariate smoothers can be constructed easily. The basis is *isotropic* so smoothing is treated the same in all directions. So if one had, a bivariate smooth of temperature and time, a one degree change in temperature would equate to a one second change in time, which is an odd assumption to make. In the more general case where units are not alike, we can use *tensor products* to combine two or more univariate smooths into a more complex basis. Each component can be made up from a different basis, playing to their particular strengths.
+TPRS are also defined for any number of predictors, so multivariate smoothers can be constructed easily. The basis is *isotropic*; smoothness is treated the same in all directions. If one wanted a bivariate smooth of temperature and time a one degree change in temperature would equate to a one second change in time with an isotropic smooth, which is an odd assumption to make. In the more general case where units are not alike, we can use *tensor products* to combine two or more univariate marginal smooths into a more complex basis. Each marginal smooth can use a different type of basis, as wanted or dictated by the data[^1]
```

Comment:
should this be a colon rather than semi-colon?

Reply:
Yes. Will correct this.